mio/stream_daq.py

```diff
@@ -359,8 +282,19 @@ def _fpga_recv(
                             buf_stop + len(self.preamble),
                         )
                     try:
+                        header, serial_buffer = BufferFormatter.bytebuffer_to_ndarrays(
```

Comment:
i think we *might* take a tiny perf hit here by moving the few array operations that are done in this method into the read loop, but i also ultimately think the multiprocessing apart from the read loop is overkill that makes working with this class harder than it needs to be, so i plan on splitting it up anyway and this is also fine to me if it doesn't break stuff for y'all

Reply:
Yes, I haven't extensively tested, but hopefully, it's not critical as it's only some partially vectorized numpy operations.